disnake/webhook/async_.py

```diff
@@ -121,6 +121,8 @@ async def request(
         except KeyError:
             self._locks[bucket] = lock = asyncio.Lock()
 
+        headers["User-Agent"] = USER_AGENT
```

Comment:
This will always override the user-agent set in user-provided `ClientSession`s

Reply:
that was intentional, as user provided sessions aren't going to provide our user agent by default 😔

Reply:
I'd like to propose only setting the user agent header if its not set here. Seems like a good middle ground without removing peoples existing functionality which does actually use user agents which could be somewhat breaking to end users in some cases

Reply:
Thing is, it is always set. We don't know if user provided sessions provided a user agent that complies with the discord api, nor do we know if the user provided a user agent. aiohttp sets a user agent by default, so we would have the wrong user agent if a user creates their own session with the defaults.

Reply:
I agree with @Skelmis here. While a one-size-fits-all solution might not be easy, indiscriminately overwriting user-agents doesn't sound like a good idea, even if it's only an edge case.
If the session has a user-agent header, that should always be kept intact, even if non-compliant with the discord api (which isn't our responsibility).

Reply:
As you can see with [`622dba8` (#959)](https://github.com/DisnakeDev/disnake/pull/959/commits/622dba8ba786495fc30eb42ffdc9d572590418c5) that is why this was not implemented.

Reply:
Honestly, looking at that I can't see a strong reason why it wasn't implemented purely based on the code outside of it possibly needing a `None` check within the `get() = ...` clause

Reply:
?

The aiohttp implementation is doable but I'd still rather not use it.


The requests implementation is where the hacky workaround is.